packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -2091,4 +2091,39 @@ void main() {
     final EditableText editableText = tester.firstWidget(find.byType(EditableText));
     expect(editableText.cursorColor, const Color(0xFFF44336));
   });
+
+  testWidgets('iOS shows selection carets', (WidgetTester tester) async {
```

Comment:
'Shows' might be a bit optimistic :D since you're not fundamentally checking that it's visible.
Maybe iOS selection handles aren't faded away by default